keps/sig-scheduling/20190926-default-even-pod-spreading.md

```diff
@@ -113,87 +116,82 @@ As a workload author, I want to spread the workload in the cluster, but:
-| Y                 | N              | -                    | No    | -                             |
+When the feature gate is enabled:
 
-Selecting `SelectorSpreadingPriority` but not `EvenPodsSpreadPriority` in a policy is an invalid
-configuration, because the latter is a requirement for the former.
+- K8s will set Default `topologySpreadConstraints` and remove `DefaultPodTopologySpread` from the
+k8s providers (`DefaultProvider` and `ClusterAutoscalerProvider`). The
+[Default](#default-constraints) will have a similar effect.
+- When a policy is used, `SelectorSpreadingPriority` will map to `PodTopologySpread`.
+- When setting plugins in the Component Config API, plugins are added as requested. Since an
+  operator is manually enabling the plugins, we assume they are aware of their intentions.
 
 ### Risks and Mitigations
 
-`EvenPodsSpreadPriority` has some overhead and we currently ensure that pods that don't use the
-feature get minimally affected. After Default `topologySpreadConstraints` is rolled out,
-all pods will run through the algorithms.
-However, we should ensure that the running overhead is not significantly higher than
-`SelectorSpreadingPriority` with k8s Default `topologySpreadConstraints`.
+The `PodTopologySpread` plugin has some overhead compared to other plugins. We currently ensure that
+pods that don't use the feature get minimally affected. After Default `topologySpreadConstraints`
+is rolled out, all pods will run through the plugin.
+We should ensure that the running overhead is not significantly higher than
+`DefaultPodTopologySpread` with the k8s Default.
 
 ## Design Details
 
 ### API
 
-A new structure called `TopologySpreadConstraint` is introduced to `KubeSchedulerConfiguration`.
+A new structure `Args` is introduced in `pkg/scheduler/framework/plugins/podtopologyspread`.
+Values are decoded from the `pluginConfig` slice in the kube-scheduler Component Config and used in
+`podtopologyspread.New`.
 
 ```go
-type KubeSchedulerConfiguration struct {
-	....
-	// TopologySpreadConstraints defines topology spread constraints to be applied to pods
+type Args struct {
+	// DefaultConstraints defines topology spread constraints to be applied to pods
 	// that don't define any in `pod.spec.topologySpreadConstraints`. Pod selectors must
 	// be empty, as they are deduced from the resources that the pod belongs to
 	// (includes services, replication controllers, replica sets and stateful sets). 
 	// If not specified, the scheduler applies the following default constraints:
 	// <default rules go here. See next section>
 	// +optional
-	TopologySpreadConstraints []corev1.TopologySpreadConstraint
-	....
+	DefaultConstraints []corev1.TopologySpreadConstraint
 }
 ```
 
-Note the use of `k8s.io/api/core/v1.TopologySpreadConstraint`. During validation,
-we verify that selectors are not defined.
+Note the use of `k8s.io/api/core/v1.TopologySpreadConstraint`. During validation, we verify that
```

Comment:
Just for curiosity, are you using some tool to auto-wrap the lines?

Reply:
Not really. I'm just (sometimes) diligent about it. My IDE just has the visual reference.